### PR TITLE
End session event on browser window

### PR DESCRIPTION
**Pull request by juturu**

### CODE REVIEW

File: atom/browser/api/atom_api_window.cc [hunk 1]
```diff
@@ -191,6 +191,12 @@ void Window::OnWindowClosed() {
       FROM_HERE, GetDestroyClosure());
 }
 
+#if defined(OS_WIN)
+void Window::OnWindowEndSession() {
+  Emit("endsession");
```
Comment (kevinsawicki) [inline atom/browser/api/atom_api_window.cc]: I think this event name should be `session-end` to match the other "end" events we currently have, `drag-end` and `scroll-touch-end`

Comment (juturu) [inline atom/browser/api/atom_api_window.cc]: Thanks @kevinsawicki. Updated the event name.

**Comment by kevinsawicki** [comment]
Thanks for adding this @juturu 👍 🏁

**Comment by akihitofujiwara** [comment]
I would appreciate if you could answer it.

How to show dialog before Windows shutdown?

Please feel free to ask me if you have any questions.

**Comment by juturu** [comment]
@akihitofujiwara please file a new issue with the scenario you are looking to support. Reason new issue is, i don't think this event will be sufficient to do what you want to do.

**Comment by akihitofujiwara** [comment]
@juturu 
Thank you for your support!
I got it.

**Pull request merged**